app/lib/lti_advantage/params.rb

```diff
@@ -0,0 +1,58 @@
+module LtiAdvantage
+  # This is for extracting data from the lti jwt in more human-readable ways
+  class Params
+    attr_reader :token
+
+    def initialize(lti_token)
+      @token = lti_token
+    end
+
+    def launch_context
+      # This is an array, I'm not sure what it means to have more than one
+      # value. In courses and accounts there's only one value
+      context = context_data["type"][0]
+
+      case context
+      when LtiAdvantage::Definitions::COURSE_CONTEXT then "COURSE"
+      when LtiAdvantage::Definitions::ACCOUNT_CONTEXT then "ACCOUNT"
+      else "UNKNOWN"
+      end
+    end
+
+    def context_id
+      context_data["id"]
+    end
+
+    def context_data
+      token[LtiAdvantage::Definitions::CONTEXT_CLAIM]
+    end
+
+    def course_id
+      value = custom_params["canvas_course_id"]
+      if value != "$Canvas.course.id"
```

Comment:
If the value is "$Canvas.course.id" did something get misconfigured? Should we throw an exception if that is the value?

Reply:
If canvas doesn't have a value to substitute then this is what it sends. So if you launch from account navigation for example. They do the same thing with old lti launches.